include/xtl/xvariant_impl.hpp

```diff
@@ -267,9 +267,13 @@ namespace std {
 #define MPARK_CPP14_CONSTEXPR
 #endif
 
-#if __has_feature(cxx_exceptions) || defined(__cpp_exceptions) || \
-    (defined(_MSC_VER) && defined(_CPPUNWIND))
-#define MPARK_EXCEPTIONS
+#if !defined(MPARK_NO_EXCEPTIONS) && \
+    (__has_feature(cxx_exceptions) || defined(__cpp_exceptions) || \
+    defined(__EXCEPTIONS) || (defined(_MSC_VER) && defined(_CPPUNWIND)))
+// Exceptions are enabled.
+#else
+// Exceptions are disabled.
+#define MPARK_NO_EXCEPTIONS
```

Comment:
It would be better to keep the old logic as much as possible, that is, define `MARK_EXCEPTIONS` when exceptions are enabled and test it after. The reason is this file is a copy - paste from mpark/variant, and when upgrading we want to avoid reviewing a diff. Besides he could add other exception guards in the implementation that could be easily missed. I know this would make use of two "redundant" macros (`MPARK_NO_EXCEPTIONS` and `MPARK_EXCEPTIONS`) but that would ease future integration.

Reply:
Yeah, ok, I can revert that. I just thought that making the NO_EXCEPTIONS case consistent was a win.